src/connection.ts

```diff
@@ -191,11 +198,39 @@ export class Connection extends JSForceConnection {
     try {
       this.setApiVersion(await this.retrieveMaxApiVersion());
     } catch (err) {
+      if (err.name === ORG_NOT_FOUND_ERROR_NAME) {
+        throw err; // throws on DNS connection errors
+      }
       // Don't fail if we can't use the latest, just use the default
       this.logger.warn('Failed to set the latest API version:', err);
     }
   }
 
+  /**
+   * Verify that instance has a reachable DNS entry, otherwise will throw error
+   */
+  public async isResolvable(): Promise<boolean> {
+    if (!this.options.connectionOptions?.instanceUrl) {
+      throw new SfdxError('Connection has no instanceUrl', 'NoInstanceUrl', [
+        'Make sure the instanceUrl is set in your command or config',
+      ]);
+    }
+    const resolver = await MyDomainResolver.create({
+      url: new URL(this.options.connectionOptions.instanceUrl),
+      timeout: Duration.seconds(10),
+      frequency: Duration.seconds(10),
+    });
+    try {
+      await resolver.resolve();
+      return true;
+    } catch (e) {
+      throw new SfdxError('The org cannot be found', ORG_NOT_FOUND_ERROR_NAME, [
+        'Verify that the org still exists',
```

Comment:
Add hint regarding possibility of domain name rename.

Reply:
What would you say for them to do here?

"If you deployed or updated the org's My Domain, update your instanceUrl". <-- I'm not sure how to do that...editing auth files directly?  

You could modify `SFDX_INSTANCE_URL`...not sure how to apply it to a specific org  in a non-local-project context ( this bug is originally coming from`force:org:list`)

I don't think we have a clean command to modify an instance on an existing username auth file, right?

Reply:
I think just the hint that the domain may have changed an logout/auth again.

BTW, where are these hints showing up? I did not see it in the sample output you shared.

Reply:
> I think just the hint that the domain may have changed and logout/auth again.

Cool, will do.

> BTW, where are these hints showing up? I did not see it in the sample output you shared.

I guess that's up to whoever is using this.  In this bug, `force:org:list` applies `error.name` into the table.  I guess we *could* modify that command's output over in toolbelt, but it's not really part of the bug.

The hints are there because lots of commands call Connection.create() and might not be handling errors, in which case the hint would go all the way to the user (if I understand `sfdxError` properly).

Reply:
Makes sense.